src/bot.ts

```diff
@@ -237,7 +238,9 @@ export class Bot<
     async init() {
         if (!this.isInited()) {
             debug("Initializing bot");
-            const me = await this.api.getMe();
+            this.mePromise ??= withRetries(() => this.api.getMe());
+            const me = await this.mePromise;
+            this.mePromise = undefined;
```

Comment:
> Why reset `this.mePromise` to undefined here? No reason for this that I can see.

There's not much of a reason apart from cleaning up after yourself, but on the other hand there's also not a reason why we should *not* set it to `undefined` again.

> Feels like this should be:
> 
> ```ts
>         if (!this.isInited()) {
>             debug("Initializing bot");
>             this.mePromise ??= withRetries(() => this.api.getMe());
>             this.me = await this.mePromise;
>         }
>         else debug("Bot info was set manually by now, will not overwrite");
> ```

This code is vulnerable to a race condition. It can occur if the library consumer assigns a value to `bot.botInfo` after calling `bot.init`. Depending on the order to the promises resolving, either the fetched bot information will be assigned to `bot.me`, or the manually set one. The extra `if` statement fixes this.

Reply:
Unsetting `mePromise` will be necessary if we want to allow userland code to manually retry when `withRetries` throws.

Otherwise, retries will simply try to await on the earlier rejected promise with no room for recovery.

Reply:
Good point! In fact, any subsequent attemts to initialise would've thrown the same error over and over again. 2f62c76 fixed this.